packages/client/modules/teamDashboard/components/AgendaItem/AgendaItem.tsx

```diff
@@ -153,15 +157,107 @@ const AgendaItem = (props: Props) => {
         gotoStageId(stageRes.stage.id)
       }
     }
-    RemoveAgendaItemMutation(atmosphere, {agendaItemId}, {meetingId})
+    if (submitting) return
+    submitMutation()
+    RemoveAgendaItemMutation(atmosphere, {agendaItemId}, {meetingId, onError, onCompleted})
   }
 
-  const getIcon = () => {
-    if (pinned && isHovering) return <SvgIcon alt='unpinIcon' src={unpinIcon} />
-    else if (!pinned && !isHovering) return <Avatar hasBadge={false} picture={picture} size={24} />
-    else return <SvgIcon alt='pinnedIcon' src={pinIcon} />
+  const getIcons = () => {
+    if (endedAt) {
+      if (pinned) {
+        if (isHovering) return <SvgIcon alt='unpinIcon' src={unpinIcon} />
+        else return <SvgIcon alt='pinnedIcon' src={pinIcon} />
+      } else {
+        if (isHovering) return <SvgIcon alt='pinnedIcon' src={pinIcon} />
+        else return <Avatar hasBadge={false} picture={picture} size={24} />
+      }
+    } else {
+      if (isHovering) return <IconButton icon='more_vert' />
+      if (!pinned) return <Avatar hasBadge={false} picture={picture} size={24} />
+      else return <SvgIcon alt='pinnedIcon' src={pinIcon} />
+    }
   }
 
+  const getMenu = () => {
+    return (
+      <Menu ariaLabel={'Select the action for the agenda item'} {...menuProps}>
+        {pinned ? (
+          <MenuItem
```

Comment:
+1 alternatively we could create an array of objects that contain the label, icon, and onClick function for each item. We could then map through the array and use the label as a key. That way, we wouldn't repeat the menuItem logic and we'd reduce the size of this already large file.